inc/html.class.php

```diff
@@ -3925,7 +3925,7 @@ static function printCleanArray($tab, $pad = 0, $jsexpand = false) {
                   }
                } else {
                   if (is_object($val)) {
-                     print_r($val);
+                     echo (string) $val; // It is expected that the object implements __toString()
```

Comment:
Thinking about it, maybe could we add an interface for Queries objects, and add a test on the interface implementation  before the `is_object` (and keep `print_r` for this case).

Reply:
I agree

As the involved classes are used as substring of a final SQL query string, I suggest one of the following names
- QueryFragmentInterface
- QueryStringInterface

Reply:
Or maybe just test for the `__toString` method? 4ve never tried, dunno if that works :)

Reply:
if will work
```php
Interactive mode enabled

php > class foo { }
php > class bar { public function __toString() { return ''; }}
php > var_dump(method_exists('foo', '__toString'));
php shell code:1:
bool(false)
php > var_dump(method_exists('bar', '__toString'));
php shell code:1:
bool(true)
```

Reply:
this is simpler, and is future proof IMHO.

Reply:
:+1: